airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DefaultTyperDeduper.java

```diff
@@ -122,31 +127,49 @@ public void prepareTables() throws Exception {
     overwriteStreamsWithTmpTable = ConcurrentHashMap.newKeySet();
     LOGGER.info("Preparing tables");
 
+    // This is intentionally not done in parallel to avoid rate limits in some destinations.
     prepareSchemas(parsedCatalog);
-    final Set<CompletableFuture<Optional<Exception>>> prepareTablesTasks = new HashSet<>();
-    for (final StreamConfig stream : parsedCatalog.streams()) {
-      prepareTablesTasks.add(prepareTablesFuture(stream));
-    }
-    CompletableFuture.allOf(prepareTablesTasks.toArray(CompletableFuture[]::new)).join();
-    reduceExceptions(prepareTablesTasks, "The following exceptions were thrown attempting to prepare tables:\n");
+
+    // TODO: Either the migrations run the soft reset and create v2 tables or the actual prepare tables.
```

Comment:
(nonblocking, just getting clarification) the only problematic migrator is the SnowflakeV2TableMigrator, right? (b/c it triggers a soft reset)

and all the other migrators are fine, b/c they just modify the raw tables? (v1v2 migrators create new v2 raw tables; bigqueryv2migrator just recreates the raw table)

I can fix this up in my branch - since I'm giving Migrations the ability to trigger a soft reset.

Reply:
I have decided to not try and fix this yet :sob: